sync/notify/bugs.py

```diff
@@ -159,34 +159,160 @@ def for_sync(sync, results):
                 continue
 
             product, component = component.split(" :: ")
-            summary, comment = bug_data(sync, test_results)
+            summary, comment = bug_data(sync,
+                                        test_results,
+                                        results.treeherder_url,
+                                        results.wpt_sha)
             bug_id = make_bug(summary, comment, product, component, [sync.bug])
             rv[bug_id] = [item + (link_status,) for item in test_results]
 
     return rv
 
 
-def bug_data_crash(sync, test_results):
-    summary = "New wpt crashes from PR %s" % sync.pr
-
-    comment = """The following tests have crashes in the CI runs for wpt PR %s:
-%s
-
-(getting the crash signature into these bug reports is a TODO; sorry)
-
-These updates will be on mozilla-central once bug %s lands.
-
-""" % (sync.pr,
-       detail_part(None, test_results, None, "head", False),
-       sync.bug)
-
-    comment += postscript
+class LengthCappedStringBuilder(object):
+    def __init__(self, max_length):
+        """Builder for a string that must not exceed a given length"""
+        self.max_length = max_length
+        self.data = []
+        self.current_length = 0
+
+    def append(self, other):
+        len_other = len(other)
+        if len_other + self.current_length > self.max_length:
+            return False
+        self.data.append(other)
+        self.current_length += len_other
+        return True
+
+    def has_capacity(self, chars):
+        return self.current_length + chars <= self.max_length
+
+    def get(self):
+        return "".join(self.data)
+
+
+def split_id(test_id):
```

Comment:
This might deserve tests. because I didn't really understood which cases you were trying to handle with `#` and `?`

```python
>>> split_id('a/b/c/d.html')
('a', 'b', 'c', 'd.html')
>>> split_id('a/b/c/d.html#foo')
('a', 'b', 'c', 'd.html#foo')
>>> split_id('a/b/c/#foo/d.html')
('a', 'b', 'c', '#foo/d.html')
```

Also there is the module urlparse that will do this for you too. 

```
>>> import urlparse
>>> urlparse.urlparse('a/b/c/d.html#foo')
ParseResult(scheme='', netloc='', path='a/b/c/d.html', params='', query='', fragment='foo')
>>> urlparse.urlparse('a/b/c/#foo/d.html')
ParseResult(scheme='', netloc='', path='a/b/c/', params='', query='', fragment='foo/d.html')
>>> urlparse.urlparse('a/b/c/d.html#foo').path
'a/b/c/d.html'
```

but not sure if it's what you want.

Reply:
I thought about using urlparse, but I'm not sure it works exactly like I want; it's important not to change the string at all, so what I really want is to just split on `/` in the path part and keep any hash or fragment part exactly the same as the input. So `foo/bar/baz.html?foo=/bar#baz` comes out as `["foo", "bar", "baz.html?foo=/bar#baz"]`. *Probably* `urlparse and then rejoining the last path part with any query or hash components does the right thing, but it's more clear that it's definitely doing the right thing if you explictly just take pieces of the input string and don't do a reconstruction step later.